### DISCUSSION TITLE
Bidirectional communication between Jetson Nano and ESP32

### DISCUSSION THREAD

**Discussion by FrankSpalteholz** [Rubberazer/JETGPIO#15] [Q&A] [answered]
As the title goes is it possible to establish a back and forth l2c communication between a Jetson Nano and an ESP32 esperif microcontroller with this library? Thank you!

**Comment by Rubberazer** [comment, marked as answer]
My library allows the Jetson to be i2c master but not slave, so with my lib the Jetson would be the one sending/receiving data to/from the ESP32, you can both "read" and "write". 
  
Now if with bidirectional what you mean is for the ESP32 to be able to initiate the communication and not only the Jetson, you would need an extra pin (GPIO) on the ESP32 that would become 1 or 0 if the ESP32 wants to start the conversation, you would connect this pin to another pin on the Jetson that would be " listening" (GPIO input) and when a change is detected, the Jetson would send/receive data to/from the ESP32 via i2c. This can be implemented with my library and it is very common using i2c, usually to indicate that new data is available on the slave side or the like.

Hope this answers your question.

**Reply by FrankSpalteholz** [reply]
Thank you **very much** for that complete and superfast answer! So it's exactly fitting my application, means Jetson as master and ESP32 as slave. Until now there is no need for switching the roles, but very good to know that with some extended work on top it's also possible. Many thanks for your work!

**Reply by Rubberazer** [reply]
No worries, check the i2c examples and also the docs: [rubberazer.github.io/JETGPIO](https://rubberazer.github.io/JETGPIO) there are function descriptions there.
